coss/home/models.py

```diff
@@ -19,6 +20,96 @@ class Meta:
 
 
 @register_snippet
+class Heading(models.Model):
+    title = models.CharField(max_length=150)
+    text = fields.RichTextField(blank=True, default='',)
+    cta_text = models.CharField(verbose_name='Text', blank=True, default='', max_length=50)
+    cta_link = models.URLField(verbose_name='Link', blank=True, default='')
+
+    panels = [
+        MultiFieldPanel([
+            FieldPanel('title'),
+            FieldPanel('text'),
+        ], heading='Content'),
+        MultiFieldPanel([
+            FieldPanel('cta_text'),
+            FieldPanel('cta_link'),
+        ], heading='Call to Action')
+    ]
+
+    def clean(self):
+        """Override clean method."""
+
+        # If there is a cta link and not a text, raise an error
+        if self.cta_link and not self.cta_text:
+            raise ValidationError({'cta_text': 'You need to provide a text for the cta link.'})
+
+        # If there is a cta text and not a link, raise an error
+        if self.cta_text and not self.cta_link:
+            raise ValidationError({'cta_link': 'You need to provide a link for the cta text.'})
+
+    def __str__(self):
+        return self.title
+
+
+@register_snippet
+class BottomBanner(models.Model):
+    title = models.CharField(blank=True, default='', max_length=50)
+    text = models.CharField(blank=True, default='', max_length=100)
+    cta_left_text = models.CharField(verbose_name='Text', blank=True, default='',
```

Comment:
Do we need this in order to tackle the issues mentioned in the PR? If not, let's drop them. I believe our action plan should be first to finish all the templates that we have for phase 1 and to only add any missing functionality. When this work is complete we can start moving things around and refactor them.

For example, ideally we want a block of code that contains only a cta link and can be re-used in multiple places in a template. I would prefer to go down this road first and if it's not possible to achieve this, then revert to using multiple entries for the same field.

Reply:
Moreover, it's better to tackle any issues/bugs separately from a refactor.

Reply:
The reason I did this here is because that banner is displayed at least on two different templates. Do you prefer I replicate all these fields in two different models for now?

Reply:
Yes! We can design the refactor the coming week so that we can get as much as we can from reusable entities that can be abstract from the design.